### PR TITLE
[ISSUE #673] Add unit tests for TopicController.

### PR DESCRIPTION
**Pull request by zhangjidi2016**
## What is the purpose of the change

#673 

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Comment by zhangjidi2016** [comment]
I used Java mock framework to add unit tests for Controller layer interface. I submitted the UT of TopicController first, please review whether this meets the requirements first. If so, I will continue to submit according to Controller later. @vongosling

File: rocketmq-console/src/test/java/org/apache/rocketmq/console/controller/BaseControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    @Before
+    public void beforeInit() throws Exception {
+        MockitoAnnotations.initMocks(this);
+        autoInjection();
+        createMockMvc();
+    }
+
+    protected void mockRmqConfigure() {
+        when(rMQConfigure.getAccessKey()).thenReturn("12345678");
```
Comment (yuz10) [inline rocketmq-console/src/test/java/org/apache/rocketmq/console/controller/BaseControllerTest.java:57]: It seems that this function can be deleted, it have no connection with the tests

Comment (zhangjidi2016) [reply rocketmq-console/src/test/java/org/apache/rocketmq/console/controller/BaseControllerTest.java:57]: image
If the unit test wants to cover all rows of this method, it need to mock the values of accessKey and secretKey not to be null

**Review by yuz10 — APPROVED** [review]
look good to me

File: rocketmq-console/src/test/java/org/apache/rocketmq/console/BaseTest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            if (MockUtil.isSpy(destObj)) {
+                // 递归处理
+                processInjection(destObj, container, localFields);
+            }
+        }
+    }
+
+    protected void processInjection(Object target, final Object localDest, final List<Field> localFields) {
+        final List<Field> fields = getAllFields(target.getClass());
+        // 比对注入
```
Comment (vongosling) [inline rocketmq-console/src/test/java/org/apache/rocketmq/console/BaseTest.java:44]: Chinese comments will not be allowed to pass, you could check the checkstyle.

**Comment by chengshanbin** [comment]
LGTM

File: rocketmq-console/src/test/java/org/apache/rocketmq/console/BaseTest.java (hunk 2 above)
Comment (vongosling) [reply rocketmq-console/src/test/java/org/apache/rocketmq/console/BaseTest.java:44]: Any ack ?

Comment (zhangjidi2016) [reply rocketmq-console/src/test/java/org/apache/rocketmq/console/BaseTest.java:44]: OK，I will recheck this commit.

**Review by vongosling — APPROVED** [review]

**Pull request merged**